### PR TITLE
boards: arm: thingy91_nrf9160: Guard calls to nrf_modem_at_printf

### PR DESCRIPTION
**Pull request by simensrostad**
Guard calls to nrf_modem_at_printf by returning early if a modem
firmware update has been performed. If modem DFU has been performed
calls to nrf_modem_at_ is expected to fail.

Closes CIA-471

### CODE REVIEW

**Review by lemrey — APPROVED** [review]
Looks fine, perhaps we can remove the problematic assertion too.

**Review by jtguggedal — APPROVED** [review]

File: boards/arm/thingy91_nrf9160/board_nonsecure.c [hunk 1]
```diff
@@ -29,11 +30,23 @@ static int thingy91_magpio_configure(void)
...
 
+	err = nrf_modem_lib_get_init_ret();
+	if (err < 0) {
+		LOG_ERR("nrf_modem_lib_get_init_ret failed, error: %d", err);
+		return err;
+	} else if (err > 0) {
+		LOG_WRN("A modem firmware upgrade has been performed, reboot is expected.");
+
+		/** If a modem firmware upgrade has been performed, calls to nrf_modem_at_
+		 *  is expected to fail until a reboot has been carried out.
```
Comment (jayteemo) [inline boards/arm/thingy91_nrf9160/board_nonsecure.c:41]: `are expected`

**Review by jayteemo — APPROVED** [review]

File: boards/arm/thingy91_nrf9160/board_nonsecure.c (hunk 1 above)
Comment (nvlsianpu) [reply boards/arm/thingy91_nrf9160/board_nonsecure.c:41]: @simensrostad Will you correct? Or merge as is.

Comment (simensrostad) [reply boards/arm/thingy91_nrf9160/board_nonsecure.c]: Fixed the comment.

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**